hydra-node/test/Hydra/Chain/Direct/TxSpec.hs

```diff
@@ -141,6 +148,58 @@ spec =
                         & counterexample (show e)
 
+    describe "commitTx" $ do
+      prop "Validate blueprint and commit transactions" $ \utxoWithWitnesses -> do
+        forAll arbitrary $ \chainContext -> do
+          let ChainContext{networkId, ownVerificationKey, ownParty, scriptRegistry} =
+                chainContext{ownVerificationKey = getVerificationKey commitSigningKey, networkId = testNetworkId}
+          let utxoToCommit = utxoFromTxOutWithWitness utxoWithWitnesses
+          let blueprintTx' = mkBlueprintTx utxoWithWitnesses
+          let tx' =
+                commitTx
+                  networkId
+                  scriptRegistry
+                  (mkHeadId Fixture.testPolicyId)
+                  ownParty
+                  utxoToCommit
+                  blueprintTx'
+                  (healthyInitialTxIn, toUTxOContext healthyInitialTxOut, verificationKeyHash ownVerificationKey)
+          let signedCommitTx = signTx commitSigningKey tx'
+          let blueprintTx = toLedgerTx blueprintTx'
+          let blueprintBody = blueprintTx ^. bodyTxL
+          let tx = toLedgerTx tx'
+          let commitTxBody = tx ^. bodyTxL
+          let containsScript u = any (isJust . witness . snd) (UTxO.pairs u)
+
+          let transactionEvaluates (transaction, lookupUTxO) =
+                case evaluateTx transaction lookupUTxO of
+                  Left err ->
+                    property False
+                      & counterexample ("Transaction: " <> renderTxWithUTxO lookupUTxO transaction)
+                      & counterexample ("Phase-1 validation failed: " <> show err)
+                  Right redeemerReport ->
+                    all isRight (Map.elems redeemerReport)
+                      & counterexample ("Transaction: " <> renderTxWithUTxO lookupUTxO transaction)
+                      & counterexample ("Redeemer report: " <> show redeemerReport)
+                      & counterexample "Phase-2 validation failed"
+          let commitUTxO =
+                UTxO.singleton (healthyInitialTxIn, toUTxOContext healthyInitialTxOut)
+                  <> utxoToCommit
+                  <> registryUTxO scriptRegistry
+
+          checkCoverage $
+            conjoin
+              [ transactionEvaluates (blueprintTx', utxoToCommit)
+              , transactionEvaluates (signedCommitTx, commitUTxO)
+              , blueprintTx ^. isValidTxL === tx ^. isValidTxL
+              , blueprintTx ^. auxDataTxL =/= tx ^. auxDataTxL
+              , property (length (blueprintBody ^. inputsTxBodyL) <= length (commitTxBody ^. inputsTxBodyL))
+              , property (length (blueprintBody ^. outputsTxBodyL) <= length (commitTxBody ^. outputsTxBodyL))
+              , property (length (blueprintBody ^. referenceInputsTxBodyL) <= length (commitTxBody ^. referenceInputsTxBodyL))
+              , property (length (blueprintBody ^. reqSignerHashesTxBodyL) <= length (commitTxBody ^. reqSignerHashesTxBodyL))
+              ]
```

Comment:
Must: also check that lower/upper bounds are retained

As described in the feature:
> Anything that does not conflict with the requirements of a valid head commit transaction should be kept. For example:
>    - A Hydra commitTx does not need to have some specific validity range, so any lower or upper bound is also present on the resulting commit tx draft
>    - Any inputs, already present are also present on the resulting commit tx draft

Of course, this requires the property test to not only work on `utxoWithWitnesses`, but using a suitable tx generator.